server/revision/revision.service.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+const { Revision, Sequelize } = require('../shared/database');
+
+const { Op } = Sequelize;
+
+function getByEntity(query, options) {
+  const { repositoryId, entity, entityId } = query;
+  const where = {
+    repositoryId,
+    entity,
+    state: { id: entityId }
+  };
+  return Revision.findAll({ ...options, where });
+}
+
+async function getPublished(query, options) {
+  const { repositoryId, activityIds, entity, entityIds = [], publishedOn } = query;
+  const removes = await Revision.findAll({
+    attributes: ['state'],
+    where: {
+      repositoryId,
+      entity,
+      operation: 'REMOVE',
+      state: { activityId: { [Op.in]: activityIds } },
```

Comment:
* Just a note that activityIds might be tricky for custom containers with nested sub-containers

Reply:
We could recursively get the graph connected to the outline activity and query `REMOVE` revisions based on IDs from that graph. Thoughts?